File: src/etch/bindings/java/transport/ServerFactory.java
package etch.bindings.java.transport;

import etch.bindings.java.support.DeliveryService;

/**
 * Interface to use for constructing new server instances by
 * TransportHelper.
 */
public interface ServerFactory
{
	/**
	 * @param d the delivery service to use with the new server.
	 * @throws Exception
	 */
	public void newServer( DeliveryService d ) throws Exception;
}

File: src/etch/bindings/csharp/transport/ServerFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using Etch.Support;
using Etch.Msg;

namespace Etch.Transport
{
    /// <summary>
    /// Interface to use for constructing new server instances by 
    /// TransportHelper
    /// </summary>
    public interface ServerFactory
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="d"> the delivery service to be used with the new service</param>
        void NewServer( DeliveryService d );

       
    }
}
